catalog/catalog.yaml

```diff
@@ -226,6 +226,22 @@ components:
       type: object
       description: Configuration for Basic Auth. Optional.
       properties:
+        apiKeyLike:
```

Comment:
seems like `apiKeyFormat` and `apiKeyField` are both fields that only make sense when `apiKeyLike` is true, it might be better for us to have a nested structure then, I also have some suggestions on naming. 

e.g. 

```
...
properties:
    apiKeyAsBasicOpts:
       type: object
       description: when this object is present, it means that this provider uses Basic Auth to actually collect an API key
       properties:
           fieldUsed:
              type: string
              enum: [username, password]
              example: username
              description: whether the API key should be used as the username or password.
              x-go-type-skip-optional-pointer: true
            keyFormat: // alternative names: keyTemplate, keyTransformation, keyFormat is fine too.
           ...
```